stdlib/public/core/StringStorage.swift

```diff
@@ -121,8 +121,13 @@ private func determineCodeUnitCapacity(_ desiredCapacity: Int) -> Int {
   let minCap = 1 + Swift.max(desiredCapacity, _SmallString.capacity)
   _internalInvariant(minCap < 0x1_0000_0000_0000, "max 48-bit length")
 
+#if _runtime(_ObjC)
+  let storageHeaderSize = 32
+  let capacity = malloc_good_size(minCap + storageHeaderSize)
```

Comment:
We will then ask for another 8 bytes for the pointer, so wouldn't this under-utilize the bucket?